README.md

```diff
@@ -1,36 +1,149 @@
 ![NodeJs Support](https://img.shields.io/badge/nodejs-%3E=8.17.0-green)
 ![NPM Support](https://img.shields.io/badge/npm-%3E=6.14.15-green)
 
-# Xenissuing
+# XenIssuing
 
-This SDK comprises of the following modules :
+This SDK comprises the following module:
 
-- XenIssuing: this module handles encryption between XenIssuing and your Web application.
+- **XenIssuing**: Handles encryption between XenIssuing and your web application.
 
 ## SecureSession
 
-SecureSession is a module to help you set up encryption between XenIssuing and your application.
+`SecureSession` is a module that helps you set up encryption between XenIssuing and your application.
 
 ### Requirements
 
-To be able to use Xenissuing, you will need to use a private key provided by Xendit.
+To use XenIssuing, you need a **public key** provided by Xendit (usually extracted from your private key).
 
 It includes several methods:
 
-- `getKey` will encrypt a session key randomly generated used for asymmetric encryption with Xenissuing.
-- `encrypt` would be used when setting sensitive data.
-- `decryptCardData` would be used whenever receiving sensitive card data from Xenissuing.
+- `getKey`: Generates and encrypts a random session key using the provided public key. This session key is used for asymmetric encryption with XenIssuing.
+- `encrypt`: Used to encrypt sensitive data before sending it.
+- `decryptCardData`: Used to decrypt sensitive card data received from XenIssuing.
 
-### Usage
+---
+
+### 1. Create Secure Session
 
 ```node
-import XenIssuing from "@xendit/xenissuing-web";
+import XenIssuing from "@xendit/xenissuing-web"
+
+const pubkey = "-----BEGIN PUBLIC KEY-----${key}-----END PUBLIC KEY-----`"
+
+const secureSession = XenIssuing.createSecureSession(pubkey) // add sessionKey as parameter to test
+const sessionId = secureSession.getKey() // to be used for API calls(eg. get card pan,cvv,etc)
+
+const { secret, iv } = apiResponse.data
+const decryptedData = secureSession.decryptCardData(iv, secret)
+```
+
+### 2. Get Card PAN
+
+```
+import axios from 'axios';
+
+const secureSession = XenIssuing.createSecureSession(pubkey);
+const authHeader = 'auth';
+const sessionId = secureSession.getKey();
+
+const config = {
+  method: 'get',
+  url: `https://your-url/pan?session_id=${sessionId}`,
```

Comment:
@UzaiSindiko , `your-url` is a bit misleading, you can just replace it with `base-url`
```suggestion
  url: `https://base-url/pan?session_id=${sessionId}`,
```